SharedPackages/DataBrokerProtectionCore/Sources/DataBrokerProtectionCore/UIWeb/DBPUICommunicationModel.swift

```diff
@@ -180,6 +196,34 @@ public struct DBPUIDataBroker: Codable, Hashable {
     }
 }
 
+public extension DBPUIDataBroker {
+    init(fromDataBroker dataBroker: DataBroker, withDate date: Date? = nil) {
+        name = dataBroker.name
+        url = dataBroker.url
+        self.date = date?.timeIntervalSince1970
+        parentURL = dataBroker.parent
+        optOutUrl = dataBroker.optOutUrl
+    }
+
+    init(fromMirrorSite mirrorSite: MirrorSite, parentBroker: DataBroker, withDate date: Date? = nil) {
+        name = mirrorSite.name
+        url = mirrorSite.url
+        self.date = date?.timeIntervalSince1970
+        parentURL = parentBroker.url
+        optOutUrl = parentBroker.optOutUrl
+    }
+}
+
+public extension Array where Element == DBPUIDataBroker {
+    func uniqued() -> Self {
+        self.reduce(into: [DBPUIDataBroker]()) { result, element in
+            if !result.contains(where: { $0.url == element.url }) {
+                result.append(element)
+            }
+        }
+    }
+}
```

Comment:
This looks like O(n^2) to me, how about using a Set of seen values similar to https://github.com/apple/swift-algorithms/blob/0e939e7bb92d5305022f0f981973c2de5ee543b4/Sources/Algorithms/Unique.swift#L115-L127? 

Granted we have a few data brokers at best, but I guess for mobile every little helps.

Reply:
My initial thought was to use the algorithms package itself which would have been nice, I can't quite remember what the impediment was there. I think we should probably start using it, so I'm gonna give it another shot

Reply:
Hmm, I remember what it was now, for some reason DBPUIDataBroker hashes based on name, not url, so I can imagine I didn't want to mess with that. 
@quanganhdo I think the name should be just as much as a unique identifier though?

Reply:
@THISISDINOSAUR `url` isn’t used to be a thing, I think that’s why we settled on `name`

https://github.com/duckduckgo/apple-browsers/blob/main/SharedPackages/DataBrokerProtectionCore/Sources/DataBrokerProtectionCore/Model/DataBroker.swift#L65-L73

Reply:
Ah, that's good to know. So are you happy that uniquing based on the name isn't going to cause any problems?

Reply:
Uniquing based on name shouldn’t be problem. The `update_embedded.sh` script already prevents duplicate names (https://github.com/duckduckgo/apple-browsers/blob/main/macOS/scripts/update_embedded.sh#L17-L36) when bundling new brokers.